### ISSUE TITLE
False positive when setting a variable within a condition

### ISSUE THREAD

**Issue by ben-manes** [uber/NullAway#97]
In this case the code is derived from `j.u.c.atomic.Striped64` so the style is retained. If setting the variable is extracted to `buffers = table` prior to the condition, then this error does not appear.

```
error: [NullAway] dereferenced expression buffers is @Nullable
      if (((buffers = table) != null) && ((n = buffers.length) > 0)) {
```

**Comment by msridhar** [comment]
Interesting!  I'll look into whether we can support this case

**Comment by msridhar** [comment]
@ben-manes I can't repro this one based on your example.  Here's what I wrote:
```java
  public void assignmentExpression2() {
    Object[] a1 = null;
    Object[] a2 = new Object[3];
    int n = -1;
    if (((a1 = a2) != null) && ((n = a1.length) > 0)) {
      System.out.println("hi");
    }
  }
```
NullAway doesn't give any warnings for this code.  Do you have a more self-contained repro example?

**Comment by ben-manes** [comment]
Does this fail for you?

```java
public void assignmentExpression2() {
  @Nullable String[] a0 = null;
  String[] a1;
  if (((a1 = a0) != null) && (a1.length > 0)) {}
}
```

```
error: [NullAway] dereferenced expression a1 is @Nullable
    if (((a1 = a0) != null) && (a1.length > 0)) {}
                                  ^
    (see http://t.uber.com/nullaway )
```

**Comment by msridhar** [comment]
Ok, yup, that does it.  This version also fails:
```java
  public void assignmentExpression2(@Nullable String[] a0) {
    String[] a1;
    if (((a1 = a0) != null) && (a1.length > 0)) {}
  }
```
Thanks!
